web/packages/teleterm/src/ui/DocumentTerminal/useDocumentTerminal.ts

```diff
@@ -389,6 +389,29 @@ function createCmd(
     };
   }
 
+  if (doc.kind === 'doc.gateway_kube') {
+    const gateway = clustersService.findGatewayByConnectionParams(
+      doc.targetUri,
+      ''
+    );
+    if (!gateway) {
+      // This shouldn't happen as DocumentGatewayCliClient doesn't render DocumentTerminal before
+      // the gateway is found. In any case, if it does happen for some reason, the user will see
+      // this message and will be able to retry starting the terminal.
+      throw new Error(`No gateway found for ${doc.targetUri}`);
+    }
+
+    const env = tshdGateway.getCliCommandEnv(gateway.gatewayCliCommand);
```

Comment:
We might want to leave a comment that this line is why this stuff actually works – the env here returns `KUBECONFIG` which is used in the terminal and the KUBECONFIG points at the gateway running underneath.